pkg/process/checks/net.go

```diff
@@ -99,7 +99,11 @@ func (c *ConnectionsCheck) Init(syscfg *SysProbeConfig, hostInfo *HostInfo, _ bo
 	c.networkID = networkID
 	c.processData = NewProcessData(c.config)
 	c.dockerFilter = parser.NewDockerProxy()
-	c.serviceExtractor = parser.NewServiceExtractor(c.sysprobeYamlConfig)
+	var (
+		enabled               = c.sysprobeYamlConfig.GetBool("system_probe_config.process_service_inference.enabled")
+		useWindowsServiceName = c.sysprobeYamlConfig.GetBool("system_probe_config.process_service_inference.use_windows_service_name")
+	)
+	c.serviceExtractor = parser.NewServiceExtractor(enabled, useWindowsServiceName)
```

Comment:
No need to use var here or in other places. You can use `:=` instead. For example:
```
enabled := c.sysprobeYamlConfig.GetBool("system_probe_config.process_service_inference.enabled")
```

Reply:
Why are we changing this? Passing the system probe config allowed the service extractor to figure out what configs it needed to be initialized and encapsulated the behavior.

Reply:
We wanted to be able to set the values of the ServiceExtractor arbitrarly and not be tied to a specific type of config. This is because the process check did not have access to the system-probe config. We discussed it in this thread @wiyu https://dd.slack.com/archives/C0201FP7SPM/p1707423196154039

Reply:
This would be cleaner in a function to encapsulate the behavior.